### PR TITLE
Feature/new laser

### PR DESCRIPTION
**Pull request by SigveBergslien**

### CODE REVIEW

File: measure/modules/draw/module.ts [hunk 1]
```diff
@@ -165,28 +165,54 @@ export class DrawModule extends BaseModule {
...
+        const closed = settings?.closed ?? true;
+        const angles = settings?.angles ?? true;
+        const generateLengthLabels = settings?.generateLengthLabels ?? false;
+        const decimals = settings?.decimals ?? 3;
+        let generateSlopeLables = false;
+        if (settings?.generateSlope != undefined) {
+            if (settings.generateSlope == true) {
+                generateSlopeLables = true;
+            } else if (settings.generateSlope == false) { }
+            else if (objectId && settings.generateSlope.has(objectId)) {
```
Comment (nd-novorender) [inline measure/modules/draw/module.ts:181]: Can `objectId` be 0?

File: measure/modules/draw/module.ts [hunk 2]
```diff
@@ -165,28 +165,54 @@ export class DrawModule extends BaseModule {
...
         }
-        var closed = settings?.closed ?? true;
-        var angles = settings?.angles ?? true;
-        var generateLineLabels = settings?.generateLineLabels ?? false;
-        var decimals = settings?.decimals ?? 3;
+        const closed = settings?.closed ?? true;
+        const angles = settings?.angles ?? true;
+        const generateLengthLabels = settings?.generateLengthLabels ?? false;
+        const decimals = settings?.decimals ?? 3;
+        let generateSlopeLables = false;
```
Comment (nd-novorender) [inline measure/modules/draw/module.ts:176]: For consistency we could use "Labels" instead of "Lables"

File: measure/modules/laser/index.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// import type { ReadonlyVec2, ReadonlyVec3 } from "gl-matrix";
```
Comment (nd-novorender) [inline measure/modules/laser/index.ts:1]: Can we remove this file?

File: measure/modules/laser/module.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// import type { DrawObject, LaserIntersections, MeasureView, MeasureWorker, ObjectId } from "measure";
```
Comment (nd-novorender) [inline measure/modules/laser/module.ts:1]: And this one?

File: measure/worker/snaps.ts [hunk 5]
```diff
@@ -84,6 +85,120 @@ export interface PickInterface {
     snappingPoints: PickPoints[];
 }
 
+function closestCandiateOnSurfaces(pickInterface: PickInterface, position: ReadonlyVec3, tolerance: number) {
```
Comment (nd-novorender) [inline measure/worker/snaps.ts:88]: typo in "Candiate"

File: measure/worker/snaps.ts [hunk 6]
```diff
@@ -84,6 +85,120 @@ export interface PickInterface {
...
+function closestSnappingPoints(pickInterface: PickInterface, position: ReadonlyVec3, tolerance: number | undefined, getAll: boolean) {
+    let closestCandidate: { entity: MeasureEntity, connectionPoint: vec3, distance: number } | undefined = undefined;
+    for (const pointInstance of pickInterface.snappingPoints) {
+        const localPoint = vec3.transformMat4(
+            vec3.create(),
+            position,
+            pointInstance.worldToObject
+        );
+        for (const snapPoints of pointInstance.points) {
+            if (isInsideAABB(localPoint, snapPoints.aabb, tolerance ? tolerance : 0.1)) {
```
Comment (nd-novorender) [inline measure/worker/snaps.ts:163]: Can tolerance be 0? Maybe better `tolerance ?? 0.1`?

File: web_app/buffer_inspect.ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import { updateMeshHighlights } from "core3d/modules/octree/mesh";
 import { vec2, type ReadonlyVec2, vec3, type ReadonlyVec3 } from "gl-matrix";
-import type { DeviationSample, OutlineSample, PickSample } from "web_app";
+import type { DeviationSample, OutlineSample, PickSample, WasmInstance } from "web_app";
```
Comment (nd-novorender) [inline web_app/buffer_inspect.ts:3]: Just new imports without code changes - accidental?

File: measure/modules/laser/index.ts (hunk 3 above)
Comment (SigveBergslien) [reply measure/modules/laser/index.ts:1]: This exposes the parametric laser. While uncomment and tag as internal while in development

**Review by nd-novorender — APPROVED** [review]

**Review by nd-novorender — APPROVED** [review]

**Pull request merged**